pkg/argo-actions/download_type_instances.go

```diff
@@ -49,7 +60,18 @@ func (d *Download) Do(ctx context.Context) error {
 			return fmt.Errorf("failed to find TypeInstance with ID %q", config.ID)
 		}
 
-		data, err := yaml.Marshal(typeInstance.LatestResourceVersion.Spec.Value)
+		typeInstanceData := DownloadTypeInstanceData{
```

Comment:
Unfortunately, I found out that the Helm upgrade won't work with this approach:
https://github.com/capactio/capact/blob/1c1917afd53f28fcaa6e28831d989738c0a329af/pkg/runner/helm/upgrade.go#L106

As you can see, we load there and input TypeInstance and try to unmarshal it.

@pkosiec do you have an opinion about it? where this should be handled:
1. directly in this `loadHelmReleaseData` method
2. or in Implementation manifests where this will be "unpacked"
3. other idea

IMO 1st is ok.

Reply:
I agree, first option is OK 👍

Reply:
I fix it in [this](https://github.com/capactio/capact/pull/657/commits/e902da7e1c6b62a4576e9dfff0c75d64b9d35977) commit.